src/main/java/com/puppycrawl/tools/checkstyle/checks/blocks/RightCurlyCheck.java

```diff
@@ -455,15 +465,15 @@ private static Details getDetailsForIfElse(DetailAST ast) {
 
         /**
          * Collects validation details for CLASS_DEF, METHOD DEF, CTOR_DEF, STATIC_INIT,
-         * INSTANCE_INIT and ANNOTATION_DEF.
+         * INSTANCE_INIT, ANNOTATION_DEF and ENUM_DEF.
          * @param ast a {@code DetailAST} value
          * @return an object containing all details to make a validation
          */
         private static Details getDetailsForOthers(DetailAST ast) {
             DetailAST rcurly = null;
             final DetailAST lcurly;
             final int tokenType = ast.getType();
-            if (tokenType == TokenTypes.CLASS_DEF || tokenType == TokenTypes.ANNOTATION_DEF) {
+            if (isDefinitionToken(tokenType)) {
```

Comment:
Once `DEF_TOKENS` will be a Set, this can be just `DEF_TOKENS.contains(tokenType)` (or `binarySearch`)
Code readability is more important then nano optimizations.

Reply:
Another way is to use `Arrays.stream`:
```
        private static boolean isDefinitionToken(int tokenType) {
            return Arrays.stream(DEF_TOKENS).anyMatch(token -> token == tokenType);
        }
```

Reply:
@sd1998 ping
Please tell us about your plans. Are you still on vacation or are you going to return to this issue?

Reply:
@pludov Yes I will the make the changes by tomorrow. Sorry for the delay.

Reply:
Done